src/gui/editors/AutomationEditor.cpp

```diff
@@ -443,18 +443,20 @@ void AutomationEditor::mousePressEvent( QMouseEvent* mouseEvent )
 		// Get the time map of current pattern
 		timeMap & tm = m_pattern->getTimeMap();
 
-		// Check if we are clicking over a node (it will be tm.end() if not)
-		timeMap::iterator clickedNode = getNodeAt(mouseEvent->x(), mouseEvent->y());
-
 		m_mouseDownLeft = (mouseEvent->button() == Qt::LeftButton);
 		m_mouseDownRight = (mouseEvent->button() == Qt::RightButton);
 
+		timeMap::iterator clickedNode;
```

Comment:
It looks like `clickedNode` is used in every switch statement, only difference is third parameter for if it's an `outValue`. Maybe we can move these lines to the declaration and decide if it's an `outValue` there?
```suggestion
		// If the event is for an outValue
		bool isOutValue = (
			m_editMode == DRAW_OUTVALUES
			|| (m_editMode == ERASE && m_mouseRightDown)
		);
		timeMap::iterator clickedNode = getNodeAt(mouseEvent->x(), mouseEvent->y(), isOutValue);
```

Reply:
True! That would save some lines, I'll change it